src/parser.rs

```diff
@@ -7458,4 +7458,11 @@ mod tests {
             ))
         );
     }
+
+    #[test]
+    fn test_update_in_with_subquery() {
+        let sql = r#"WITH "result" AS (UPDATE "Hero" SET "name" = 'Captain America', "number_of_movies" = "number_of_movies" + 1 WHERE "secret_identity" = 'Sam Wilson' RETURNING "id", "name", "secret_identity", "number_of_movies") SELECT * from "result""#;
```

Comment:
It's better to have a test case that either asserts that the query roundtrips correctly, so that we can make sure the `Display` half of the implementation works too.